### PR TITLE
Add SMI Conformance Docs

### PR DESCRIPTION
**Pull request by pottekkat**
**Description**

This PR adds docs for "SMI Conformance with Meshery" including the following:

- What is SMI conformance and how Meshery fits in it.
- Running SMI conformance tests through Meshery UI
- Running SMI conformance tests through Meshery CLI
- Running SMI conformance in CI/CD pipelines. Needs to be updated as https://github.com/meshery/meshery-osm/pull/102 is merged and tested.

**Notes for Reviewers**

[Changes made here 1](https://6115885518d8bd3361ec74f7--meshery-docs-preview.netlify.app/functionality/service-mesh-interface)

[Changes made here 2](https://6115885518d8bd3361ec74f7--meshery-docs-preview.netlify.app/guides/smi-conformance)

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 7ce230595f182eaacdf1aeb6fa783ad550886cb7 at: https://6115885518d8bd3361ec74f7--meshery-docs-preview.netlify.app

File: docs/pages/guides/smi-conformance.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+       - name: Setup MiniKube
+         uses: manusa/actions-setup-minikube@v2.4.1
+         with:
+           minikube version: 'v1.21.0'
+           kubernetes version: 'v1.20.7'
+           driver: docker
+ 
+       - name: Install OSM
+         run: |
+            curl -LO https://github.com/openservicemesh/osm/releases/download/v0.6.0/osm-v0.6.0-linux-amd64.tar. gz
```
Comment (leecalcote) [inline docs/pages/guides/smi-conformance.md:131]: Ouch. @DelusionalOptimist are we ready to replace this with `mesheryctl mesh`?

Comment (navendu-pottekkat) [reply docs/pages/guides/smi-conformance.md:131]: Waiting for the PR linked in the PR description. @leecalcote

Comment (navendu-pottekkat) [reply docs/pages/guides/smi-conformance.md:131]: And once the PR is merged, we need to modify the action as well to use `mesheryctl mesh`. The issue currently being Meshery is not able to deploy the latest OSM.

Comment (DelusionalOptimist) [reply docs/pages/guides/smi-conformance.md:131]: hehe. yeah just an inch away :pinching_hand:

**Comment by navendu-pottekkat** [comment]
@leecalcote Could you review the rest of the docs? I will update the docs for the action once it is updated.

**Review by leecalcote — APPROVED** [review]

**Review by navendu-pottekkat — COMMENTED** [review]
CITY

**Comment by l5io** [comment]
🚀 Preview for commit 5226c6f0c630bfe13a32c9fac7b793d186ccaad4 at: https://61174ed1056e6a38f21cada8--meshery-docs-preview.netlify.app

**Pull request merged**